foodsaving/unsubscribe/utils.py

```diff
@@ -0,0 +1,82 @@
+from django.contrib.contenttypes.models import ContentType
+from django.core import signing
+
+from foodsaving.conversations.models import ConversationThreadParticipant, ConversationParticipant
+from foodsaving.pickups.models import PickupDate
+
+
+def generate_token(user, group=None, conversation=None, thread=None):
+    data = {'u': user.id}
+    if group:
+        data.update({
+            'g': group.id,
+            'gn': group.name,
+        })
+    if conversation:
+        data.update({'c': conversation.id})
+    elif thread:
+        if not thread.is_first_in_thread():
+            raise Exception('your thread is not a thread!')
+        data.update({'t': thread.id})
+    return signing.dumps(data)
+
+
+def parse_token(token):
+    return signing.loads(token)
+
+
+def unsubscribe_from_conversation(user, conversation):
+    ConversationParticipant.objects.filter(
+        user=user,
+        conversation=conversation,
+    ).update(
+        email_notifications=False,
+    )
+
+
+def unsubscribe_from_thread(user, thread):
+    ConversationThreadParticipant.objects.filter(
+        user=user,
+        thread=thread,
+    ).update(
+        muted=True,
+    )
+
+
+def unsubscribe_from_all_conversations_in_group(user, group):
```

Comment:
I imagine that we easily forget to update this when we add another conversation type...

Reply:
I implemented it using a generic approach now. In the case of adding a new thing that has a conversation, the thing needs to have a `group` attribute. It will error if it does not.